integration_test/networks.test.ts

```diff
@@ -0,0 +1,485 @@
+      }).to.throw('Unrecognized network 421613.')
+      expect(() => {
+        getL1Network(42170)
+      }).to.throw('Unrecognized network 42170.')
+    })
+    it('should throw an error if passed an L1 Chain ID not in the Networks list', () => {
+      // Chain ID 137 is Polygon
+      expect(() => {
+        getL1Network(137)
+      }).to.throw('Unrecognized network 137.')
+    })
+  })
+
+  context('getL2Network', () => {
+    it('should return an L2 network if passed an L2 Chain ID in the Networks list', () => {
+      const getL2NetworkResult42161 = getL2Network(42161)
+      const getL2NetworkResult421611 = getL2Network(421611)
+      const getL2NetworkResult421613 = getL2Network(421613)
+      const getL2NetworkResult42170 = getL2Network(42170)
+
+      expect(getL2NetworkResult42161).to.deep.equal(l2Networks[42161])
+      expect(getL2NetworkResult421611).to.deep.equal(l2Networks[421611])
+      expect(getL2NetworkResult421613).to.deep.equal(l2Networks[421613])
+      expect(getL2NetworkResult42170).to.deep.equal(l2Networks[42170])
+    })
+    it('should return an L2 network if passed an L2 Provider with a network in the Networks list', async () => {
+      const { provider: l2Provider42161 } = createProviderMock(42161)
+      const { provider: l2Provider421611 } = createProviderMock(421611)
+      const { provider: l2Provider421613 } = createProviderMock(421613)
+      const { provider: l2Provider42170 } = createProviderMock(42170)
+
+      const getL2NetworkResult42161 = await getL2Network(l2Provider42161)
+      const getL2NetworkResult421611 = await getL2Network(l2Provider421611)
+      const getL2NetworkResult421613 = await getL2Network(l2Provider421613)
+      const getL2NetworkResult42170 = await getL2Network(l2Provider42170)
+
+      expect(getL2NetworkResult42161).to.deep.equal(l2Networks[42161])
+      expect(getL2NetworkResult421611).to.deep.equal(l2Networks[421611])
+      expect(getL2NetworkResult421613).to.deep.equal(l2Networks[421613])
+      expect(getL2NetworkResult42170).to.deep.equal(l2Networks[42170])
+    })
+    it('should throw an error if passed an L1 Chain ID', () => {
+      expect(() => {
+        getL2Network(1)
+      }).to.throw('Unrecognized network 1.')
+      expect(() => {
+        getL2Network(1338)
+      }).to.throw('Unrecognized network 1338.')
+      expect(() => {
+        getL2Network(4)
+      }).to.throw('Unrecognized network 4.')
+      expect(() => {
+        getL2Network(5)
+      }).to.throw('Unrecognized network 5.')
+    })
+    it('should throw an error if passed an L2 Chain ID not in the Networks list', () => {
+      // Chain ID 420 is Optimism Goerli
+      expect(() => {
+        getL2Network(420)
+      }).to.throw('Unrecognized network 420.')
+    })
+  })
+
+  context('addCustomNetwork', () => {
+    it('should add the custom L1 and L2 networks to the l1Networks and l2Networks objects respectively', () => {
+      const customL1Network: L1Network = {
```

Comment:
We repeat these network configs many times in this file, we could have a function that creates these configs and just call it each time - will be less verbose.